### PR TITLE
add sometrue and alltrue

### PR DESCRIPTION
**Pull request by guilhermeleobas**
This PR implements the NumPy functions `np.sometrue` and `np.alltrue`.

### CODE REVIEW

**Comment by esc** [comment]
@guilhermeleobas thanks for this, I have queued it for review! As usual, thanks for helping to improve Numba. 😄

File: numba/targets/arraymath.py [hunk 1]
```diff
@@ -573,6 +573,13 @@ def flat_all(a):
     return flat_all
 
 
+@overload(np.alltrue)
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: These need adding to the supported functions list.

**Comment by guilhermeleobas** [comment]
Oops, completely forgot about `numpysupported.rst`

**Comment by stuartarchibald** [comment]
Thanks for the fixes. Marking for merge.

**Comment by stuartarchibald** [comment]
Before merge we should check if this API is actually still part of NumPy.

**Comment by guilhermeleobas** [comment]
`np.sometrue` exists but it's not documented. The same is true for `np.alltrue`.

**Comment by stuartarchibald** [comment]
Seems like this is potentially scheduled for deprecation over in NumPy https://github.com/numpy/numpy/issues/14584. Following a long discussion with other Numba core developers about supporting features that are not-in-use/potentially going to be deprecated, the consensus was that it was probably best to not replicate/maintain features subject to such conditions. Are you ok with this outcome @guilhermeleobas? Thanks for your understanding, and again, asking these questions is likely to help NumPy long term too.

**Comment by guilhermeleobas** [comment]
Sure, no problem.

**Pull request closed without merging**